src/core/agent/services/multiSigService.ts

```diff
@@ -615,6 +617,46 @@ class MultiSigService extends AgentService {
     };
   }
 
+  async readyMultisigRotation(identifierId: string) {
+    const metadata = await this.identifierStorage.getIdentifierMetadata(
+      identifierId
+    );
+
+    const multiSig = await this.props.signifyClient
+      .identifiers()
+      .get(metadata.signifyName);
+
+    const members = await this.props.signifyClient
+      .identifiers()
+      .members(metadata?.signifyName);
+
+    const nextSequence = (Number(multiSig.state.s) + 1).toString();
+
+    const rmids = members?.rotation;
+
+    const rstates: any[] = [];
+    await Promise.allSettled(
+      rmids.map(async (rotation: any) => {
+        const aid = await this.props.signifyClient
+          .keyStates()
+          .query(rotation.aid, nextSequence);
+        if (aid.done) {
+          rstates.push(aid.response);
+        }
+      })
+    );
+
+    const result: any = [];
+
+    rstates.forEach((rstate) => {
+      if (rstate.i !== multiSig.state.i) {
+        result.push(rstate);
+      }
+    });
+
+    return result;
```

Comment:
I think we can actually return the ones that are complete, and just the identifier.

So:
```
const rotated: string[] = [];
for (const rstate of r) {
  if (rstate.i === multiSig.state.i) {
    rotated.push(rstate.i);
  }
}

return result;
```

(I think for of is more clear than forEach)

Reply:
I think after we get `rstates`, `rstate.id` will be different current state of multisig
So case `rstate.i === multiSig.state.i` will not happen

Reply:
Sorry I was reviewing this too quickly before my meeting ;) You're right. Still, we can just respond with the identifiers (`.i`)

One thing though: based on my comment above - https://github.com/cardano-foundation/cf-identity-wallet/pull/592#discussion_r1686290073 - we can't fully rely on the index because as I said I could do 10 rotations of my local member identifier before we do a rotation of the multi-sig. So this won't be reliable.

I created a ticket so that we fetch the latest state and not the state based on the sequence number - but we will also need to check that the public keys of the local member identifier does not equal the public keys of the multi-sig, instead of checking the sequence numbers like you are doing here.

We can do both changes later if you want - but I think since you have time, maybe it would be better to just do the public key check now? (and in that other story, we can change the query call too)

Reply:
Is the ticket you're talking about on Keria?
So what I need to do here is just delete the `nextSequence` from `.query`, right?

Reply:
Yes, remove `nextSequence` but not until later. Our KERIA version doesn't support that correctly at the moment (there was a bug).

I'm saying right now: instead of checking `rstate.i !== multiSig.state.i` - we should check that the public keys of rstate !== public keys of multiSig.